src/DynamoCoreWpf/UI/GuidedTour/GuidesManager.cs

```diff
@@ -0,0 +1,207 @@
+        /// </summary>
+        private Guide currentGuide;
+        private UIElement mainRootElement;
+
+        /// <summary>
+        /// GuidesManager Constructor that will read all the guides/steps from and json file and subscribe handlers for the Start and Finish events
+        /// </summary>
+        /// <param name="root">root item of the main Dynamo Window </param>
+        public GuidesManager(UIElement root)
+        {
+            mainRootElement = root;
+            Guides = new List<Guide>();
+            ReadGuides("UI/GuidedTour/dynamo_guides.json");
+
+            //Subscribe the handlers when the Tour is started and finished, the handlers are unsubscribed in the method TourFinished()
+            GuideFlowEvents.GuidedTourStart += TourStarted;
+            GuideFlowEvents.GuidedTourFinish += TourFinished;
+        }
+
+        /// <summary>
+        /// This method will launch the tour when the user clicks in the Help->Interactive Guides->Guide
+        /// </summary>
+        /// <param name="tourName"></param>
+        public void LaunchTour(string tourName)
+        {
+            GuideFlowEvents.OnGuidedTourStart(tourName);
+        }
+
+        /// <summary>
+        /// This method will be executed when the OnGuidedTourStart event is raised
+        /// </summary>
+        /// <param name="args">This parameter will contain the GuideName as a string</param>
+        private void TourStarted(GuidedTourStateEventArgs args)
+        {
+            currentGuide = (from guide in Guides where guide.Name.Equals(args.GuideName) select guide).FirstOrDefault();
+            if (currentGuide != null)
+            {
+                currentGuide.Initialize();
+                currentGuide.Play();
+            }
+        }
+
+        /// <summary>
+        /// This method will be executed when the OnGuidedTourFinish event is raised
+        /// </summary>
+        /// <param name="args">This parameter will contain the GuideName as a string</param>
+        private void TourFinished(GuidedTourStateEventArgs args)
+        {
+            currentGuide = (from guide in Guides where guide.Name.Equals(args.GuideName) select guide).FirstOrDefault();
+            if (currentGuide != null)
+            {
+                foreach (Step tmpStep in currentGuide.GuideSteps)
+                {
+                    tmpStep.StepClosed -= Popup_StepClosed;
+                }
+                currentGuide.ClearGuide();
+                GuideFlowEvents.GuidedTourStart -= TourStarted;
+                GuideFlowEvents.GuidedTourFinish -= TourFinished;
+            }
+
+        }
+
+        /// <summary>
+        /// This method will read all the guides information from a json file located in the same directory than the DynamoSandbox.exe is located.
+        /// </summary>
+        /// <param name="jsonFile">Full path of the json file location</param>
+        private void ReadGuides(string jsonFile)
```

Comment:
Not a fan of huge function body, can we break this one up?

Reply:
@QilongTang I've broken up this function in several ones in the next commit: